# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Mohamed-mahmoudd/ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

 I chose the Refresh / Content Opportunity Scoring lane because I think it can help us know which pages need to be updated first.

 The goal is to give these pages a better chance to get more visits instead of updating pages randomly.


In [ ]:
import os, subprocess

REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if not os.path.isdir(REPO_DIR):
    subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)

os.chdir(REPO_DIR)
print(os.getcwd())

/content/flyrank-ml-internship-starter/flyrank-ml-internship-starter


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Question : which pages should be reviewed first based on the available data?

Decision : Decide which pages should have higher priority for updating.

Action : The content team updates the pages with the highest priority first.

Cost of a wrong recommendation : If the model gives a high score to a page that does not need an update, the team may waste time updating it.
If it gives a low score to a page that really needs an update, we may lose the chance to improve its performance.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Declining pages:", (df["trend_direction"] == "down").sum())
print("Average CTR by position tier:")
print(df.groupby("position_tier")["ctr"].mean().round(4))

Total pages: 30000
Declining pages: 16262
Average CTR by position tier:
position_tier
deep        0.1502
page_1      0.6525
page_3_5    0.2225
striking    0.3232
top_3       1.4836
Name: ctr, dtype: float64


The starter dataset contains 30,000 pages,and 16,262 pages are marked as declining, which means there are many pages that may need review. Also, the average CTR changes a lot between position tiers, showing that page performance is different across search positions. These observations make this lane worth exploring.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

I cannot say that updating a page will definitely improve its performance.

I can only say that the data suggests some pages may have a higher priority for review based on the available features.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.